docs/reviewing.md

```diff
@@ -105,20 +105,39 @@ def _configure_cmake(self):
 
-The contents of `test_package.c` or `test_package.cpp` should be as minimal as possible, including a few headers at most with simple instantiation of objects to ensure linkage
-and dependencies are correct.
+The contents of `test_package.c` or `test_package.cpp` should be as minimal as possible, including a few headers at most with simple
+instantiation of objects to ensure linkage and dependencies are correct. Any build system can be used to test the package, but
+CMake or Meson are usually preferred.
 
-### Verifying Components
+### CMake targets
 
-When components are defined in the `package_info` in `conanfile.py` the following conditions are desired
+When using CMake to test a package, the information should be consumed using the **targets provided by `cmake_find_package_multi` generator**. We
+enforce this generator to align with the upcoming 
+[Conan's new `CMakeDeps` generator](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html?highlight=cmakedeps)
+and it should help in the migration (and compatibility) with Conan v2.
 
-* use `cmake_find_package` if library has an [official](cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules) CMake module emulated in the recipe.
-* use `cmake_find_package_multi` if library provides an official cmake config file emulated in the recipe. If there are more than one target, try to use all of them, or add another executable linking to the global (usually unofficial) target. There may additionally be variables that are emulated by the recipe which should be used as well. There are some ways to identify when to use it:
-    * Usually, project installs its cmake files into `package_folder/lib/cmake`. The folder is removed from package folder by calling `tools.rmdir(os.path.join(self.package_folder), "lib", "cmake")`
-    * Also, the library's CMake scripts can use [install(EXPORT ..)](https://cmake.org/cmake/help/latest/command/install.html#export) and/or may install [package configuration helpers](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html)
-      to indicate an exported CMake config file.
-    * When `self.cpp_info.filenames["cmake_find_package_multi"]`, `self.cpp_info.names["cmake_find_package_multi"]` are declared
-* otherwise, use [cmake generator](https://docs.conan.io/en/latest/reference/generators/cmake.html) to not suggest an unofficial cmake target in test package.
+In ConanCenter we try to mimic the names of the targets and the information provided by CMake's modules and config files that some libraries
+provide. If CMake or the library itself don't enforce any target name, the ones provided by Conan should be recommended. The minimal project
+in the `test_package` folder should serve as an example of the best way to consume the package, and targets are preferred over raw variables.
+
+This rule applies for the _global_ target and for components ones. The following snippet should serve as example:
```

Comment:
I would suggest:

- cmake_find_package_multi if there is an official config file
- cmake_find_package if there is an official Find module file and no official config file
- pkg_config if there is an official .pc file and no official cmake Find/config files.
- cmake_find_package_multi otherwise

Reply:
I would also warn against relying on unofficial CMake targets in recipes themselves, it can break at anytime while CONAN_PKG targets are robust.

Reply:
I'ld also prefer using `CONAN_PKG::` targets or similar. Something in a conan namespace.
I would consider it unfriendly if some package manager would start dictating cmake targets to my project.

At least, there should be some comment added that a target is conan specific and prone to breakage.

Reply:
To give a heads-up of what is happening in Conan 2.0:

- The only CMake generator that remains is ``CMakeDeps`` and ``CMakeToolchain``. They generate ``pkg::pkg`` and ``pkg::component`` targets, not any CONAN_LIB or CONAN_PKG targets.
- The idea is to provide a fully transparent integration with CMake, the push for that has been massive over the years. 
- Also, to provide 1 integration for CMake, supporting many ones is is big overhead in support in maintenance, we need to converge to 1 solution

Reply:
Will CMakeDeps be able to generate both module and configs?
If not, that will break a lot of recipes at cci

Reply:
Yes, it is already able in Conan 1.40

Reply:
They tagged us in a PR with the feature asking for feedback :)

Reply:
Leaving the links here, it maybe useful:
- codebase PR: https://github.com/conan-io/conan/pull/9455
- cocs PR: https://github.com/conan-io/docs/pull/2209/files
- actual docs: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html#properties